### PR TITLE
chore: support graphql@16 traversal logic

### PR DESCRIPTION
**Pull request by gabrielcsapo**
# Summary

Upgrades to graphql@16 and supports the new traversal syntax.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "test:debug": "node --inspect-brk node_modules/.bin/jest --runInBand --no-watchman --config ./jest.config.cjs"
   },
   "peerDependencies": {
-    "graphql": "^0.8.0 || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0 || ^16.0.0"
```
Comment (asakusuma) [inline package.json:19]: do we actually need to drop other graphql versions to make the other changes in this PR?

Comment (gabrielcsapo) [reply package.json:19]: https://github.com/graphql/graphql-js/releases/tag/v16.0.0 sadly we can't do this easily.

Comment (gabrielcsapo) [reply package.json:19]: we can support 15 it seems, so 15 and 16 would be the graphql dependencies.

File: yarn.lock [hunk 2]
```diff
@@ -2131,7 +2131,7 @@ graphql-tag@^2.11.0:
   dependencies:
     tslib "^2.1.0"
 
-graphql@^14.7.0:
+graphql@^14.5.8:
```
Comment (asakusuma) [inline yarn.lock:2134]: shouldn't this be graphql 16 now?

Comment (gabrielcsapo) [reply yarn.lock]: updated.

**Pull request merged**